index.js

```
@@ -27,51 +27,92 @@ app.use(bodyParser.json());
-			.then(response => response.json())
-			.then(data => {
-				console.log('D::', data)
-
-				if(data.ok) {
-					res.end();
-				} else {
-					//TODO: throw Error here and focus the error handling throughout the app.
-					res.send('We have issues processing the request, contact the app admin');
-				}
-			})
-			.catch(err => console.log(err));
+		.then(response => response.json())
+		.then(data => {
+			console.log('D::', data);
+
+			if (data.ok) {
+				res.end();
+			} else {
+				//TODO: throw Error here and focus the error handling throughout the app.
+				res.send('We have issues processing the request, contact the app admin');
+			}
+		})
+		.catch(err => console.log(err));
 });
 
 app.post('/submit', async (req, res) => {
 	const response = JSON.parse(req.body.payload);
-	const actionType = response.type;
 	const viewID = response.view.id;
 	const user = response.user.name;
 	const values = response.view.state.values;
+	const private_metadata = JSON.parse(response.view.private_metadata);
+	const response_url = private_metadata.response_url;
 	const submission = Input.submit(viewID, user, values);
 
+	const response_msg_process = Structure.processing(submission.productName);
+	postData(response_url, response_msg_process);
+
 	if(submission) {	
-		return await Sheet.write(submission, data => {
+		Sheet.write(submission, data => {
 			Input.deleteView(viewID);
-			return res.json(Structure.confirm(data, res));
+
+			const response_msg_confirm = Structure.confirm(data);
+			postData(response_url, response_msg_confirm);
 		});
 	}
+
+	return res.json(Structure.clear());
 });
 
+async function postData(url, data) {
+	const options = {
+		method: 'POST',
+		body: JSON.stringify(data),
+		headers: {
+			'Content-Type': 'application/json',
+			Authorization: 'Bearer ' + process.env.SLACK_TOKEN
+		}
+	};
+
+	return fetch(url, options)
+		.then(response => {
+			if(response.error != null ){
+				console.log('error')
```

you should throw an error so the `catch` can get it (and handle it in the catch eventually...)

---

Done